doc/development/adding_operators.rst

```diff
@@ -0,0 +1,356 @@
+            # which makes sure an error is raised if the wrong number was passed
+            return 1
+
+        @staticmethod
+        def compute_decomposition(angle, wires, do_flip):  # pylint: disable=arguments-differ
+            # Overwriting this method defines the decomposition of the new gate, as it is
+            # called by Operator.decomposition().
+            # The general signature of this function is (*parameters, wires, **hyperparameters).
+            op_list = []
+            if do_flip:
+                op_list.append(qml.PauliX(wires=wires[1]))
+            op_list.append(qml.RX(angle, wires=wires[0]))
+            return op_list
+
+        def adjoint(self):
+            # the adjoint operator of this gate simply negates the angle
+            return FlipAndRotate(-self.parameters[0], self.wires[0], self.wires[1], do_flip=self.hyperparameters["do_flip"])
+
+The new gate can now be created as follows:
+
+>>> op = FlipAndRotate(0.1, wire_rot="q3", wire_flip="q1", do_flip=True)
+>>> op
+FlipAndRotate(0.1, wires=['q3', 'q1'])
+>>> op.decomposition()
+[PauliX(wires=['q1']), RX(0.1, wires=['q3'])]
+>>> op.adjoint()
+FlipAndRotate(-0.1, wires=['q3', 'q1'])
+
+The new gate can be used with PennyLane devices. PennyLane checks with the device
+whether it supports operations using the operation name.
+
+- If the device registers support for an operation with the same name,
+  PennyLane leaves the gate implementation up to the device. The device
+  might have a hardcoded implementation, _or_ it may refer to one of the
+  numerical representations of the operator (such as `operator.matrix`).
+  
+- If the device does not register support for an operation with the same
+  name, PennyLane will automatically decompose the gate using `operator.decomposition`.
+
+.. code-block:: python
+
+    from pennylane import numpy as np
+
+    dev = qml.device("default.qubit", wires=["q1", "q2", "q3"])
+
+    @qml.qnode(dev)
+    def circuit(angle):
+        FlipAndRotate(angle, wire_rot="q1", wire_flip="q1")
+        return qml.expval(qml.PauliZ("q1"))
+
+>>> a = np.array(3.14)
+>>> circuit(a)
+-0.9999987318946099
+
+If all gates used in the decomposition have gradient recipes defined,
+we can even compute gradients of circuits that use the new gate without any extra effort.
+
+>>> qml.grad(circuit)(a)
+-0.0015926529164868282
+
+.. note::
+
+    The example of ``FlipAndRotate`` is simple enough that one could write a function
+
+    .. code-block:: python
+
+        def FlipAndRotate(angle, wire_rot, wire_flip=None, do_flip=False):
+            if do_flip:
+                qml.PauliX(wires=wire_flip)
+            qml.RX(angle, wires=wire_rot)
+
+    and call it in the quantum function *as if it was a gate*.
+    However, classes allow much more functionality, such as defining the adjoint gate above,
+    defining the shape expected for the trainable parameter(s), or specifying gradient rules.
```

Comment:
Is this ok? I also changed the title back to "adding new ops", so it is clear that this is more for devs trying to add them to PennyLane!

Reply:
Sounds much clearer! 💯